### ISSUE TITLE
Publish JSAPI types nightly/on demand

### ISSUE THREAD

**Issue by mattrunyon** [deephaven/deephaven-core#5150]
As a developer, I want JSAPI types published nightly or on demand so that when we implement a new API during a cycle, we can implement the web side without needing to wait for a full release to get types for the new API.

**Comment by niloc132** [comment]
Added @stanbrub and @devinrsmith so we can discuss considering this in a wider conversation of publishing other artifacts on a nightly basis - I believe we only ship the docs and docker images today.

**Comment by stanbrub** [comment]
There is a _docs-ci.yml_ workflow that runs a _typedocs_ step that deploys on push to deephaven-core _main_.  https://deephaven.io/core/client-api/javascript/modules/dh.html.  Is this what you mean?  Or am I missing some context?

**Comment by niloc132** [comment]
Yes, this isn't about the generated html for the typescript API, but the tarball deployed to https://www.npmjs.com/package/@deephaven/jsapi-types for use in downstream projects.
